app/flatpak-builtins-list.c

```diff
@@ -117,21 +205,27 @@ find_refs_for_dir (FlatpakDir *dir, GStrv *apps, GStrv *runtimes, GCancellable *
 }
 
 static gboolean
-print_table_for_refs (gboolean print_apps, GPtrArray * refs_array, const char *arch, GCancellable *cancellable, GError **error)
+print_table_for_refs (gboolean print_apps, GPtrArray * refs_array, const char *arch, const char *columns, GCancellable *cancellable, GError **error)
 {
-  FlatpakTablePrinter *printer = flatpak_table_printer_new ();
+  FlatpakTablePrinter *printer;
+  g_auto(GStrv) cols = NULL;
+  g_autofree int *col_idx = NULL;
+  int n_cols;
   int i;
 
-  i = 0;
-  flatpak_table_printer_set_column_title (printer, i++, _("Ref"));
-  if (opt_show_details)
+  cols = g_strsplit (columns, ",", 0);
+  n_cols = g_strv_length (cols);
+  col_idx = g_new (int, n_cols);
+  for (i = 0; i < n_cols; i++)
     {
-      flatpak_table_printer_set_column_title (printer, i++, _("Origin"));
-      flatpak_table_printer_set_column_title (printer, i++, _("Active commit"));
-      flatpak_table_printer_set_column_title (printer, i++, _("Latest commit"));
-      flatpak_table_printer_set_column_title (printer, i++, _("Installed size"));
+      col_idx[i] = find_column (cols[i], error);
+      if (col_idx[i] == -1)
+        return FALSE;
```

Comment:
Maybe put a column option splitter in the utils too

Reply:
yeah, I'll look at that